File: Java/3349. Adjacent Increasing Subarrays Detection I.java
class Solution {
    public boolean hasIncreasingSubarrays(List<Integer> nums, int k) {
        int left = 0, streak = 1;
        for (int i = 1; i < nums.size(); i++){
            if (nums.get(i) > nums.get(i - 1)) {
                streak++;
            } else {
                left = streak;
                streak = 1;
            }

            if ((streak >= 2 * k) || (streak >= k && left >= k)) {
                return true;
            }
        }

        return false;
    }
}

File: Python3/3349. Adjacent Increasing Subarrays Detection I.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        streak, left = 1, 0
        for i, n in enumerate(nums[1:], 1):
            if n > nums[i - 1]:
                streak += 1
            else:
                left = streak
                streak = 1
            
            if streak >= 2 * k or streak >= k and left >= k:
                return True

        return False